pkg/util/label.go

```diff
@@ -124,3 +125,20 @@ func RecordManagedLabels(object *unstructured.Unstructured) {
 	annotations[workv1alpha2.ManagedLabels] = strings.Join(managedKeys, ",")
 	object.SetAnnotations(annotations)
 }
+
+// DedupeAndMergeFinalizers merges the new finalizers into exist finalizers.
+func DedupeAndMergeFinalizers(existFinalizers, newFinalizers []string) []string {
+	if existFinalizers == nil {
+		return existFinalizers
+	}
+	existFinalizerSets := sets.Set[string]{}
+	existFinalizerSets.Insert(existFinalizers...)
+
+	newFinalizerSets := sets.Set[string]{}
+	newFinalizerSets.Insert(existFinalizers...)
+	newFinalizerSets.Insert(newFinalizers...)
+	if equality.Semantic.DeepEqual(existFinalizerSets, newFinalizerSets) {
+		return existFinalizers
+	}
+	return newFinalizerSets.UnsortedList()
```

Comment:
Here should return a sorted list, right?

Reply:
The order should be 

items in `exist`  --> items in `new`(but not in `exist`)

Reply:
I'll fix it, pls review again  @RainbowMango